src/containers/SearchPage/SearchPage.shared.js

```diff
@@ -496,3 +538,35 @@ export const getDatesAndSeatsMaybe = (currentParams, newParams) => {
       : { seats: null, dates: null };
   return datesAndSeatsMaybe;
 };
+
+/**
+ * Returns params for createResourceLocatorString function based on the current
+ * location and route configuration
+ * @param {*} routes current route configuration
+ * @param {*} location current ReactRouter location
+ * @returns an object with the attributes routeName and pathParams, which can then be passed
+ * as the corresponding parameters to createResourceLocatorString
+ */
+export const getResourceLocatorStringParams = (routes, location) => {
+  const testURL = new URL(`http://my.marketplace.com${location.pathname}`);
+  const matchedRoutes = matchPathname(testURL.pathname, routes);
+
+  if (matchedRoutes.length > 0) {
```

Comment:
What should happen in unhappy path? 
Return nulls, throw error, default to landing page ?